src/engine/pieces/knight.ts

```diff
@@ -1,13 +1,29 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Knight extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableMoves = new Array<Square>();
+        // let finalAvailableMoves = [];
+        let currentPosition = board.findPiece(this);
+        let oldRow = currentPosition.row;
+        let oldCol = currentPosition.col;
+
+        let ones = [-1, 1]
+        let twos = [-2, 2]
```

Comment:
It's not the most clear what `ones` and `twos` are from their names

Reply:
Fixed - not sure if it still makes enough sense though